core/src/banking_stage/consume_worker.rs

```diff
@@ -180,22 +180,26 @@ pub(crate) struct ConsumeWorkerMetrics {
 
 impl ConsumeWorkerMetrics {
     /// Report and reset metrics iff the interval has elapsed and the worker did some work.
-    pub fn maybe_report_and_reset(&self) {
-        const REPORT_INTERVAL_MS: u64 = 1000;
-        if self.interval.should_update(REPORT_INTERVAL_MS)
-            && self.has_data.swap(false, Ordering::Relaxed)
-        {
-            self.count_metrics.report_and_reset(&self.id);
-            self.timing_metrics.report_and_reset(&self.id);
-            self.error_metrics.report_and_reset(&self.id);
+    pub fn maybe_report_and_reset(&self, slot: Option<Slot>) {
+        if slot.is_some() {
```

Comment:
If I am leader for slots `[100, 101, 102, 103]`, we want to report those metrics in a timely fashion.
As is, this will only report for `103` whenever my next leader slot is. We need to handle the case where the new `slot` argument is `None` (indicating we are not leader).